components/automate-cli/pkg/verifyserver/services/fqdnservice/fqdnservice.go

```diff
@@ -0,0 +1,261 @@
+package fqdnservice
+
+import (
+	"crypto/md5"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/hex"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/constants"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/lib/logger"
+)
+
+type IFqdnService interface {
+	CheckFqdnReachability(models.FqdnRequest, string) models.FqdnResponse
+}
+
+type FqdnService struct {
+	log     logger.Logger
+	timeout time.Duration
+}
+
+func NewFqdnService(log logger.Logger, timeout time.Duration) IFqdnService {
+	return &FqdnService{
+		log:     log,
+		timeout: timeout,
+	}
+}
+
+func (fq *FqdnService) createClient(rootCert string) *http.Client {
+	fq.log.Debug("Creating Client")
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM([]byte(rootCert))
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs:    caCertPool,
+				MinVersion: tls.VersionTLS12,
+			},
+		},
+		Timeout: fq.timeout * time.Second,
+	}
+	fq.log.Debug("Client Created.")
+	return client
+}
+
+func createErrorMessage(setNodes map[string]int, reqnodes []string, isAfterDeployment bool) string {
+	temp := "["
+	for _, k := range reqnodes {
+		var val string
+		if isAfterDeployment {
+			hasher := md5.New() // nosemgrep
+			_, err := io.WriteString(hasher, k)
+			if err != nil {
+				return constants.IP_TO_HASH_FAIL_MESSAGE
+			}
+			val = hex.EncodeToString(hasher.Sum(nil))
+		} else {
+			val = k
+		}
+
+		if setNodes[val] > 0 {
```

Comment:
this is not the way of checking the key in the map

Reply:
done